src/modules/Game.class.js

```diff
@@ -6,63 +6,323 @@
+            this.score += filteredRow[j] * 2;
+            j--;
+          } else {
+            mergedRow.unshift(filteredRow[j]);
+          }
+        }
+
+        while (mergedRow.length < this.size) {
+          mergedRow.unshift(0);
+        }
+
+        this.board[i] = mergedRow;
+      }
+
+      if (!this.isBoardEqual(prevBoard, this.board)) {
+        this.addRandomNumber();
+      }
+      this.getStatus();
+    }
+  }
+
+  moveUp() {
+    if (this.status === 'playing') {
+      const prevBoard = this.board.map((row) => [...row]);
+
+      for (let col = 0; col < this.size; col++) {
+        const filteredColumn = [];
+
+        // Формуємо колону без нулів
+        for (let row = 0; row < this.size; row++) {
+          if (this.board[row][col] !== 0) {
+            filteredColumn.push(this.board[row][col]);
+          }
+        }
+
+        const mergedColumn = [];
+
+        // Об'єднуємо однакові числа
+        for (let i = 0; i < filteredColumn.length; i++) {
+          if (filteredColumn[i] === filteredColumn[i + 1]) {
+            mergedColumn.push(filteredColumn[i] * 2); // Об'єднання
+            this.score += filteredColumn[i] * 2; // Оновлюємо рахунок
+            i++; // Пропускаємо наступне число
+          } else {
+            mergedColumn.push(filteredColumn[i]);
+          }
+        }
+
+        // Додаємо нулі в кінець колони
+        while (mergedColumn.length < this.size) {
+          mergedColumn.push(0);
+        }
+
+        // Оновлюємо значення в дошці
+        for (let row = 0; row < this.size; row++) {
+          this.board[row][col] = mergedColumn[row];
+        }
+      }
+
+      // Перевіряємо, чи змінилася дошка
+      if (!this.isBoardEqual(prevBoard, this.board)) {
+        this.addRandomNumber();
+      }
+      this.getStatus();
+    }
+  }
+
+  moveDown() {
+    if (this.status === 'playing') {
+      // eslint-disable-next-line max-len
+      const prevBoard = this.board.map((row) => [...row]); // Копіюємо дошку перед змінами
+
+      for (let col = 0; col < this.size; col++) {
+        const filteredColumn = [];
+
+        // Формуємо колону без нулів
+        for (let row = this.size - 1; row >= 0; row--) {
+          if (this.board[row][col] !== 0) {
+            filteredColumn.unshift(this.board[row][col]);
+          }
+        }
+
+        const mergedColumn = [];
+
+        // Об'єднуємо однакові числа
+        for (let i = filteredColumn.length - 1; i >= 0; i--) {
+          if (filteredColumn[i] === filteredColumn[i - 1]) {
+            mergedColumn.unshift(filteredColumn[i] * 2); // Об'єднання
+            this.score += filteredColumn[i] * 2; // Оновлюємо рахунок
+            i--; // Пропускаємо наступне число
+          } else {
+            mergedColumn.unshift(filteredColumn[i]);
+          }
+        }
+
+        // Додаємо нулі на початок колони
+        while (mergedColumn.length < this.size) {
+          mergedColumn.unshift(0);
+        }
```

Comment:
This logic is similar and duplicated in every method, try to move it to a helper function and use it everywhere.